subcommand/acl-controller/command.go

```diff
@@ -88,23 +88,24 @@ func (c *Command) run() error {
 			CAPem: []byte(caCert),
 		}
 	}
-	if c.flagPartitionsEnabled {
-		cfg.Partition = c.flagPartition
-	}
 
 	consulClient, err := api.NewClient(cfg)
 	if err != nil {
 		return err
 	}
 
+	if err = c.createPartition(consulClient); err != nil {
+		return err
+	}
+
 	smClient := secretsmanager.New(clientSession, nil)
 
-	err = c.upsertConsulClientToken(consulClient, smClient)
-	if err != nil {
-		return err
+	if c.flagPartitionsEnabled {
+		cfg.Partition = c.flagPartition
```

Comment:
Since the Consul client is already initialized, does this do anything? Even if it does, I wonder if this is just something that would either cause a bug elsewhere or cover up a bug based on [the description in the code](https://github.com/hashicorp/consul/blob/main/api/api.go#L343-L345).

Reply:
Grepping shows that we specify partition in a bunch of places on requests, so I wonder if this is something that would just cover up or cause a bug.

```
controller/resource_test.go
1150:		ns, _, err := consulClient.Namespaces().List(&api.QueryOptions{Partition: p.Name})
1164:	opts := &api.QueryOptions{Partition: partition}
1196:		&api.QueryOptions{Partition: partition, Namespace: namespace})

controller/resource.go
201:	opts := &api.QueryOptions{Partition: s.Partition}
289:		&api.QueryOptions{Partition: s.Partition, Namespace: DefaultNamespace},
330:	existingNS, _, err := s.ConsulClient.Namespaces().List(&api.QueryOptions{Partition: s.Partition})
445:	opts := &api.QueryOptions{Partition: s.ServiceName.Partition, Namespace: s.ServiceName.Namespace}
```

Reply:
No, it does nothing.. The `Config` is initialized and then a copy is stored in the `Client`: https://github.com/hashicorp/consul/blob/main/api/api.go#L725. That's a good catch, I'll remove it.